datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -478,7 +481,7 @@ func (f *ChainDataFetcher) retryFunc(insert HandleChainEventFn) HandleChainEvent
 		for err := insert(event, reqType); err != nil; err = insert(event, reqType) {
 			select {
 			case <-f.stopCh:
-				return nil
+				return err
```

Comment:
Not related. to this PR, however, why did you make `retryFunc` to return function? What's the difference between this and direct execution?

Reply:
I thought the wrapping function is more flexible to pre-process or post-process of the given function `HandleChainEvent`.